src/dialects/abstract/query-generator.js

```diff
@@ -2861,6 +2861,15 @@ Only named replacements (:name) are allowed in literal() because we cannot guara
 
         return this._joinKeyValue(key, this.escape(pattern, undefined, options), comparator, options.prefix);
       }
+
+      case Op.anyKeyExists:
+      case Op.allKeysExist: {
+        if (value instanceof Utils.SequelizeMethod) {
+          return this._joinKeyValue(key, this.handleSequelizeMethod(value, undefined, undefined, options), comparator, options.prefix);
+        }
+
+        return this._joinKeyValue(key, `ARRAY[${value.map(item => this.escape(item, undefined, options)).join(', ')}]`, comparator, options.prefix);
```

Comment:
I just thought of a case we missed!

If `value` is empty, this will produce `jsonb ?| ARRAY[]`. Postgres won't be able to infer the type of the array. Luckily both `?|` and `?&` require a text array. If `value` is empty, could you make it produce `jsonb ?| ARRAY[]::text[]`? (don't forget to add a test for it). No need to add the cast the rest of the time